framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -0,0 +1,141 @@
+ *   };
+ *
+ *   return (
+ *     <Box align="left">
+ *       <CreatableSelectDropdown
+ *         standardOptions={options}
+ *         onOptionChange={handleOptionChange}
+ *         width="300px"
+ *       />
+ *       { artist && artist.value !== 'custom-option' && (
+ *         <H3 py={8}>The best artist is: { artist.label }</H3>
+ *       ) }
+ *     </Box>
+ *   );
+ * }
+ * ?)
+ */
+
+export const CreatableSelectDropdown = ({
+  standardOptions,
+  initialPlaceholder = 'Select or create...',
+  addOptionPlaceholder = 'Enter text...',
+  creationOption = {
+    label: 'Add option...',
+    value: 'custom-option',
+  },
+  onOptionChange,
+  width = '100%',
+  variant = 'outline',
+}: CreatableSelectDropdownProps) => {
+  const [ options, setOptions ] = useState(standardOptions)
+  const [ selectedOption, setSelectedOption ] = useState<Option | null>(null)
+  const [ newOptionText, setNewOptionText ] = useState('')
+  const [ newOptionPlaceholder, setNewOptionPlaceholder ] = useState(initialPlaceholder)
+  const [ menuIsOpen, setMenuIsOpen ] = useState(false)
+
+  const handleInputChange = (newValue: string) => {
+    setNewOptionText(newValue)
+  }
+
+  const handleCreateOption = () => {
+    const newOptionTextLower = newOptionText.toLowerCase()
+    const newOptionExists = options.some((option) =>
+      option.value.toLowerCase() === newOptionTextLower)
+
+    if (!newOptionExists) {
+      const newOption = { label: newOptionText, value: newOptionText }
+      setOptions((currentOptions) => [ ...currentOptions, newOption ])
+      setSelectedOption(newOption)
+      onOptionChange(newOption)
+    }
+
+    setNewOptionText('')
+    setMenuIsOpen(false)
+    setNewOptionPlaceholder(initialPlaceholder)
+  }
+
+  const handleChange = (option: Option) => {
+    if (option?.value === 'custom-option') {
+      setNewOptionPlaceholder(addOptionPlaceholder)
+      setSelectedOption(null)
+    } else {
+      setNewOptionPlaceholder(initialPlaceholder)
+      setSelectedOption(option)
+    }
+    setMenuIsOpen(false)
+    onOptionChange(option)
+  }
+
+  const handleKeyDown = (event: { key: string, preventDefault: () => void }) => {
+    if (event.key === 'Enter' && newOptionText) {
+      handleCreateOption()
+      event.preventDefault()
+    }
+  }
+
+  const customOptions = [
+    {
+      ...creationOption,
+      icon: <Icon as={ PlusSolid } color="brand" />,
+    },
+    ...options,
+  ]
+
+  return (
+    <CreatableSelect
+      chakraStyles={ {
+        container: (provided) => ({
+          ...provided,
+          width,
+        }),
+        option: (provided, { isSelected }) => ({
+          ...provided,
+          ...(isSelected && {
+            color: 'black',
+          }),
+        }),
+      } }
+      components={ customComponents }
+      options={ customOptions }
+      value={ selectedOption }
+      onChange={ (option) => option && handleChange(option) }
```

Comment:
added this inside handleChange now

    if (option === null) {
      return
    }
    
    not sure if(where you want me to use ??, since I dont want to assign a default value when dealing with null values, instead I dont want to perform the action then